packages/aeproject-cli/aeproject-init/init.js

```diff
@@ -194,15 +197,87 @@ const setupDeploy = async (shape) => {
     }
 }
 
-const setupDocker = async () => {
+const setDockerImageVersion = (pathToDockerYmlFile, dockerImage) => {
+    let doc = yaml.safeLoad(fs.readFileSync(pathToDockerYmlFile, 'utf8'));
+
+    let tokens = dockerImage.split(':');
+    let imageLiteral = tokens[0];
+
+    for (let i in doc.services) {
+        let image = doc.services[i].image;
+
+        if (image.startsWith(imageLiteral)) {
+            doc.services[i].image = dockerImage;
+        }
+    }
+
+    let yamlStr = yaml.safeDump(doc);
+    fs.writeFileSync(pathToDockerYmlFile, yamlStr, 'utf8');
+}
+
+const setupDocker = async (isUpdate) => {
     print(`===== Creating docker directory =====`);
+    
     const dockerFilesSource = `${ __dirname }${ constants.artifactsDir }/${ constants.dockerTemplateDir }`;
-    const copyOptions = {
-        overwrite: false
+    const dockerNodeYmlFileSource = `${ __dirname }${ constants.artifactsDir }/${ constants.dockerNodeYmlFile }`;
+
+    const nodeTokens = constants.aeNodeImage.split(':');
+    const compilerTokens = constants.aeCompilerImage.split(':');
+
+    let nodeVersion = [];
+    let compilerVersion = [];
+    
+    // get user's version
+    if (isUpdate) {
+
+        const aeternityNodeImageLiteral = nodeTokens[0];
+        const aeternityCompilerImageLiteral = compilerTokens[0];
+
+        try {
+            // read user's node yml
+            let userNodeYmlPath = path.join(process.cwd(), constants.dockerNodeYmlFile);
+            let doc = yaml.safeLoad(fs.readFileSync(userNodeYmlPath, 'utf8'));
+
+            for (let i in doc.services) {
+                let image = doc.services[i].image;
+
+                if (image.startsWith(aeternityNodeImageLiteral)) {
+                    let imageTokens = image.split(':')
+                    let currentVersion = imageTokens[1];
+
+                    nodeVersion.push(currentVersion);  
```

Comment:
why not getting the current version from the constants file, instead of reading the yml?